tox.ini

```diff
@@ -179,3 +179,34 @@ install_command = pip install --no-deps {opts} {packages}
 commands =
     conda list
     {env:PYTEST_COMMAND} {posargs}
+
+[testenv:pyinstaller]
+# Check that astropy can be included in a PyInstaller bundle without any issues. This
+# also serves as a test that tests do not import anything from outside the tests
+# directories with relative imports, since we copy the tests out to a separate directory
+# to run them.
+description = check that astropy can be included in a pyinstaller bundle
+changedir = .pyinstaller
+deps =
+    # The developer version of PyInstaller doesn't work yet because it
+    # requires https://github.com/pyinstaller/pyinstaller/pull/4888 -
+    # so for now we pin to the latest working commit.
+    git+https://github.com/pyinstaller/pyinstaller.git@c111ec5c#egg=PyInstaller
+    # PyInstaller is not yet compatible with Matplotlib 3.3 so we pin
+    # Matplotlib to an older version for now - see the following issue
+    # for more details: https://github.com/pyinstaller/pyinstaller/issues/5004
+    matplotlib<3.3
+extras =
+    tests
+    net
+    all
+commands =
+    pyinstaller --onefile run_sunpy_tests.py \
+                --distpath . \
+                --additional-hooks-dir hooks \
+                --exclude-module tkinter \
+                --hidden-import pytest \
+                --hidden-import pytest_doctestplus.plugin \
+                --hidden-import pytest_mpl.plugin
+
+    ./run_sunpy_tests
```

Comment:
How much of this do we need?

Reply:
What do you mean? I shall remove the comments, I forgot to do that.

Reply:
Each command, the mpl version, the pyinstaller version?

Reply:
The matplotlib issue still persists with PyInstaller so that is required.
Do we need to test pyinstaller with the dev version? Why not just use the latest working commit?

Reply:
The matplotlib 3.3 incompatibility should have been cleared up in pyinstaller/pyinstaller#5006 and then 3.4 compatibility in pyinstaller/pyinstaller#5568 which is included in PyInstaller 4.3 (which released a few days ago). If you're still seeing breakages then that's quite worrying...